deepchem/models/torch_models/gnn.py

```diff
@@ -0,0 +1,360 @@
+import torch
+from torch_geometric.nn import GINEConv, global_add_pool, global_mean_pool, global_max_pool
+from torch_geometric.nn.aggr import AttentionalAggregation, Set2Set
+from torch.functional import F
+from deepchem.models.torch_models import ModularTorchModel
+from deepchem.feat.graph_data import BatchGraphData
+
+num_atom_type = 120
+num_chirality_tag = 3
+# Relevant in future PRs
+# num_bond_type = 6
+# num_bond_direction = 3
+
+
+class GNN(torch.nn.Module):
+    """
+    GNN module for the GNNModular model.
+
+    This module is responsible for the graph neural network layers in the GNNModular model.
+
+    Example
+    -------
+    >>> from deepchem.models.torch_models.gnn import GNNModular
+    >>> from deepchem.feat.graph_data import BatchGraphData
+    >>> from deepchem.feat.molecule_featurizers import SNAPFeaturizer
+    >>> featurizer = SNAPFeaturizer()
+    >>> smiles = ["C1=CC=CC=C1", "C1=CC=CC=C1C=O", "C1=CC=CC=C1C(=O)O"]
+    >>> features = featurizer.featurize(smiles)
+    >>> batched_graph = BatchGraphData(features).numpy_to_torch(device="cuda")
+    >>> modular = model = GNNModular("gin", 3, 64, 1, "attention", 0, "last", "edge_pred")
+    >>> gnnmodel = modular.gnn
+    >>> print(gnnmodel(batched_graph)[0].shape)
+    torch.Size([23, 64])
+
+    """
+
+    def __init__(self,
+                 atom_type_embedding,
+                 chirality_embedding,
+                 gconvs,
+                 batch_norms,
+                 dropout,
+                 JK,
```

Comment:
What does JK mean? Generally better to use descriptive rather than short variable names

Reply:
Can you replace with `jump_knowledge`? `JK` is shorter but less maintainable and readable

Reply:
fixed